lib/hamt.js

```diff
@@ -528,31 +534,44 @@ Map.prototype.isEmpty = function() {
 /**
     Alter the value stored for `key` in `map` using function `f` using
     custom hash.
-    
+
     `f` is invoked with the current value for `k` if it exists,
     or no arguments if no such value exists. `modify` will always either
     update or insert a value into the map.
```

Comment:
Update the comments where required as well. The project doesn't use any sort of documentation generator currently, but the comments should be human readable and help someone browsing the code understand what is going on.